datafusion/physical-expr/src/intervals/interval_aritmetic.rs

```diff
@@ -235,18 +247,16 @@ impl Display for Interval {
 impl Interval {
     /// Creates a new interval object using the given bounds.
     ///
-    /// # Boolean intervals need special handling
+    /// As explained in [`Interval]` boolean `Interval`s are special and this
```

Comment:
Given that there are only three valid boolean intervals, and `Interval::new()` normalizes to one of those, it might make sense to make the fields of `Interval` non `pub` so that it is not possible to construct an invalid Interval

Reply:
In fact, that's exactly what I'm working on. I have planned to set the fields to private and create intervals with only `try_new`. If an interval can be created with the given parameters, it creates that interval. If the parameters cannot construct a valid interval (like lower: `[true` -  upper: `false]` for a boolean interval), it returns an error.